docs/design/fabric-v2+/fpc-validation.puml

```diff
@@ -52,88 +52,107 @@ group endorsement
     end ref
     return proposal response
 
+    alt FPC "light"
+        User1 -> ECC1 ++: invoke "validateEnclaveEndorsement" (proposal response)
+            group enclave endorsement validation
+                ECC1 -> ECC1 : <args, read/writeset, result> <- extract from transaction
+                ECC1 -> ECC1 : <Enclave_VK, Sig_Enclave, ECC_result> <- extract from result
+
+                ECC1 -> ECC1 : verify Sig_Enclave over <args, read/writeset, ECC_result> with Enclave_VK
+                ECC1 -> ECC1 : Enclave_Id <- Hash(Enclave_VK)
+                create participant ERCC order 30
+                ECC1 -> ERCC++ : queryEnclaveCredentials(CC_Id, Enclave_Id)
+                    note right
+                        If FPC enclave is registered, ERCC will
+                        return the corresponding Credentials,
+                        otherwise Credentials is empty.
+                    end note
+                return Credentials
+                ECC1 -> ECC1 : extract CC_Params from Credentials.attested_data
+                ECC1 -> ECC1 : check CC_Params matches chaincode definition
+                ECC1 -> ECC1 : check CC_Params.MSPID matches MSPID of endorser
```

Comment:
> check CC_Params.MSPID matches MSPID of endorser

Is this still the right thing to do with FPC light?

Reply:
it's misleading text. It has to be the identity of the transaction creator (which essentially represents as an (conceptual, not Fabric) endorsement of this enclave to represent its organization. Will rephrase ...

Reply:
> ECC1 -> ECC1 : check CC_Params.MSPID matches MSPID of proposal creator & creator has role Admin

I think something is wrong here. With this check we enforce that only members of the org owning the enclave and members that have admin rights can invoke FPC transactions. 

What we want:
- only admins can create fpc enclaves
- every org in the channel can invoke the fpc chaincode hosted by any org

Reply:
oops, good catch. This seems to have been wrong for quite a while already before my changes; i just was thinking of delta from your earlier review comment. Will fix ...

Reply:
Good catch. This is wrong and now thinking not only about the delta from your earlier comment, i realize that this was wrong even beforehand. CC_Params doesn't have MSPID, it's now in credentials.   So i guess one could conceptually try verify that the signature from the enclave still comes from an endorsing peer consistent with the registration. But this will be more challenging now in externalized validation and i also don't think the test is really necessary: as part of the registration an organization has taken ownership, i.e., consider itself accountable, and so any mischief of that enclave might be laid blame on the org, making it in their interest to protect the enclave (keys) well
=>
Propose to just drop that line?

Reply:
drop it!

Reply:
just pushed a fixup.  Realized there were also a few other inconsistencies which are hopefully now fixed (or at least improved ;-))